.github/workflows/generate-changelog.yaml

```diff
@@ -58,15 +58,26 @@ on:
         default: 'chore: generate changelog'
         type: string
 
+      token_app_id:
+        description: 'A GitHub App ID used to generate an access token to create a pull request.'
+        required: true
+        type: string
+
     secrets:
-      token:
-        description: 'Access token to manage the pull request.'
-        required: false
+      token_private_key:
+        description: 'A GitHub App private key used to generate an access token to create a pull request.'
 
 jobs:
   generate_changelog:
     runs-on: ubuntu-latest
     steps:
+      - name: 'Create GitHub App Token'
```

Comment:
This step generates a GitHub App access token, instead of passing in directly a Personal Access Token as we used to do. This is the recommended approach by IDX and is used in the monorepo now too.